client.go

```diff
@@ -1436,11 +1436,15 @@ func (r *Receiver) Receive(ctx context.Context) (*Message, error) {
 	msg := Message{receiver: r} // message to be decoded into
 
 	var (
-		maxMessageSize = int(r.link.maxMessageSize)
+		maxMessageSize = 0
 		messageSize    = 0
 		first          = true // receiving the first frame of the message
 	)
 
+	if r.link.maxMessageSize < math.MaxInt32 {
+		// if the max size is less than max int set, otherwise, it might as well be infinite
+		maxMessageSize = int(r.link.maxMessageSize)
+	}
```

Comment:
I think it might be better to track the message size as an `uint64` instead of considering anything greater than `math.MaxInt32` unlimited. I think it's unlikely anyone transfers messages that large over AMQP but I don't want to completely disregard an explicitly configured max size.

Reply:
rebased with uint64 support